File: c/39.c
/* 840 */
#include <math.h>
#include <stdio.h>

#define N 1000

int main(int argc, char **argv)
{
    double c;
    int a, b, pp;
    int p[N+1] = {0};
    int m = 0, mp = 0;
    for (a = 1; a < N/2; a++)
    {
        for (b = a; b < N/2; b++)
        {
            c = sqrt(a * a + b * b);
            pp = a + b + c;
            if ((int)c == c && pp <= N)
            {
                if (++p[pp] > m)
                {
                    m = p[pp];
                    mp = pp;
                }
            }
        }
    }
    printf("%d\n", mp);
    return 0;
}

File: py/39.py
# 840
import math

N = 1000
p = [0 for i in xrange(N + 1)]
for a in xrange(1, N/2):
    for b in xrange(a, N/2):
        cc = math.sqrt(a * a + b * b)
        c = int(cc)
        if c == cc and a + b + c <= N:
            p[a + b + c] += 1

m = 0
mv = 0
for i in xrange(N + 1):
    if p[i] > m:
        m = p[i]
        mv = i
print mv
